mulog-adv-console/README.md

```diff
@@ -10,6 +10,82 @@ advanced formatting options.
 
 Please see [README](../README.md#advanced-console-publisher) on main page.
 
+### ANSI color printing
+For better visual checks of the logs there is an ANSI color printer.
+
+To use it you need to setup the rules when the format applies, as well as the formatters
+
+The current setup supports 8 colors: 
+  - white
+  - black
+  - red
+  - green
+  - blue
+  - yellow
+  - magenta
+  - cyan
+
+You can add a background using `:bg-` in front of the color - e.g. `:bg-red` will make the background red.
+
+In addition to these you can use `:bright` to achieve a similar effect to bold. Use `:underline` to underline.
+
+`:inverse` will inverse the font color and the background color.
+
+#### Example for setting up the formats:
+
+There are two types of formatters:
+- `event` - when the value is found in an log item it will color the whole event
+  - when there are two or more matches it will pick the last rule it matches
+  - when there are not matches it will pick the `:default-formatter` otherwise it will print a plain string
+- `pair` - it overrides the event color and colors a specific key-value pair.
+  - when there are two or more matches it will pick the last rule
+
+
+```clojure
+(advanced-console/register-formatters
+  :http-format           {:event :yellow}
+  :event-format          {:event :green}
+  :http-error-format     {:pair :red}
+  :override-pair-format  {:pair :blue}
+  :underline-pair-format {:pair [:cyan :underline :bg-red :inverse]}
+  :default-formatter     :magenta})
+```
+
+In order to apply multiple formatting effects you need to wrap them in a vector.
+
+#### Example for setting up the rules
+
+To make use of the rules setup you need `where` to be included in your project: `[com.brunobonacci/where "0.5.5"]`
+
+```clojure
+;; notice that the matching values are used to match the formatter.
+(def format-rules
+  [(where :mulog/event-name :is? :line-test)
+   {:line-test :event-format}
+
+   (where contains? :http-test)
+   {:http-test :http-format}
```

Comment:
I don't understand what the map stands for here.

My undestanding is:
the rules are pairs for conditions followed by the format to apply  like
`[test format, test2 format2, ..., testN formatN]` (probably this should be explained in the doc)
why do you have to specify a map for the format? what does the key represent in the map?
why not simply specifying the format?
like:
``` clojure
   (where contains? :http-test)
   :http-format
```

Reply:
what is the logic behind the rule applications? **first one/last wins?**, **rules are cumulative?**

Reply:
The rules are cumulative. Will clarify this further in the readme.

Having the map makes sense for the pairs since there it's not obvious which pair the rule might apply to. 
For the event the map is not necessary.

In this case it makes sense to move specifying which pair to be colored in the formatter.